### PR TITLE
[Database metrics] Update metrics captured

### PR DESCRIPTION
**Pull request by chinandrew**
A few unrelated short changes. Hopefully not too bad to review, but let me know if it would be easier if it was split up.

Summary of Changes:
- Add a total rows metric. 
    - Add a function that queries the info and one that parses the returned bytestring. Same pattern as the database disk usage.
    - Updates the function to return metrics, also a small change here to return output and not the exit code.
    - Add/update tests
- Change memory usage from a list of usages across time to the max usage measured
- Fix some type annotations and typo docstrings

### CODE REVIEW

File: src/database_metrics/parsers.py [hunk 1]
```diff
@@ -40,7 +60,8 @@ def parse_metrics(metrics: tuple) -> dict:
     -------
     Dictionary containing final disk usage, runtime, and list of memory usage during function call.
```
Comment (krivard) [inline src/database_metrics/parsers.py:61]:
```suggestion
    Dictionary containing the number of rows in the database, final disk usage, runtime, and the maximum memory usage queried during the function call.
```

**Review by krivard — CHANGES_REQUESTED** [review]
Missing docstring update, otherwise 👍

File: src/database_metrics/parsers.py (hunk 1 above)
Comment (chinandrew) [reply src/database_metrics/parsers.py:61]: thanks for catching, changes added

**Review by krivard — APPROVED** [review]
lgtm

**Pull request merged**